website/static/js/passwordForms.js

```diff
@@ -243,15 +266,51 @@ var SignUpViewModel = oop.extend(BaseViewModel, {
 
 /** Wrapper classes */
 var ChangePassword = function(selector) {
-    $osf.applyBindings(new ChangePasswordViewModel(), selector);
+    this.ChangePasswordViewModel = new ChangePasswordViewModel();
+    $osf.applyBindings(this.ChangePasswordViewModel, selector);
+        // Necessary to prevent enter submitting forms with invalid frontend zxcvbn validation
+        $(selector).keypress(
+            event => {
+            // If the enter key is pressed to submit a form, check if the password is valid
+            if (event.which === 13) {
+                    if (!this.ChangePasswordViewModel.password.isValid()) {
+                        return false;
+                    }
+                }
+            }
+        );
 };
 
 var SetPassword = function(selector) {
-    $osf.applyBindings(new SetPasswordViewModel(), selector);
+    this.SetPasswprdViewModel = new SetPasswordViewModel();
```

Comment:
Typo here. Also--as mentioned in my previous comment--no need to assign to `this`.

```js
var viewModel = new SetPasswordViewModel();
// ...
    if (viewModel.password.isValid()) {
         // ... 
    }
```